### ISSUE TITLE
[4.0] Joomla 4.0 botched install

### ISSUE THREAD

**Issue by emuthomi** [joomla/joomla-cms#19867]
### Steps to reproduce the issue
I downloaded the J4.0 alpha release to test drive J4.0 and went straight to installation

### Expected result
I expected a swift and smooth install as usual with Joomla.

### Actual result
The first issue encountered was a max_execution_time fatal error

![screenshot-localhost-2018-03-08-12-14-16-841](https://user-images.githubusercontent.com/3242390/37152919-023858d8-22ec-11e8-86e4-3a66e7dcafbf.png)

The installer actually created some tables but didn't finalize the table creation somehow. Went to check and saw this:

![screenshot-localhost-2018-03-08-12-15-04-041](https://user-images.githubusercontent.com/3242390/37152961-2bed58c2-22ec-11e8-96e1-2b3c51a57317.png)

I then increased the max_execution_time in PHP.ini to 60 seconds and tried the install in another directory. The result:

![screenshot-localhost-2018-03-08-16-12-16-669](https://user-images.githubusercontent.com/3242390/37152994-4fe2f458-22ec-11e8-8862-ccc3c3ad2cb7.png)

### System information (as much as possible)
System: Windows NT 6.1 build 7601 (Windows 7 Professional Edition Service Pack 1) i586
Architecture: x86
PHP: Version 7.1.1
Server API: Apache 2.0 Handler

max_execution_time --- 30
max_file_uploads -------20
max_input_nesting_level ----- 64
max_input_time --------- 60
max_input_vars ------- 1000
memory_limit  -------- 128M

Joomla Version: joomla-cms-4.0.0-alpha1

### Additional comments
I suggest the addition of a "prerequisite checks screen" before the installation.
This screen will show things like the minimum php requirements and if every requirement is there before installation.

**Comment by brianteeman** [comment]
Please use the nightly build available at developer.joomla.org as the alpha is very old.

As for your comment about prerequisite checks them this does exist you just don't see it if you meet them.

I am going to close this for now but if you have the same issues with the nightly it can be reopened or you can create a new issue.

**Comment by emuthomi** [comment]
Let me try the nightly development package and test.

**Comment by emuthomi** [comment]
@brianteeman after trying the 4.0 nightly development build from here https://developer.joomla.org/nightly-builds.html, the installer ran for an insane amount of time.

I decided to give it time to complete but nearly an hour later, it was still "installing".
I got curious and checked the error logs only to find this:

```
[Thu Mar 08 16:39:55.243751 2018] [php7:error] [pid 6440:tid 1636] [client ::1:49444] PHP Fatal error:  Maximum execution time of 60 seconds exceeded in C:\\xampp71\\htdocs\\joomla4dev\\libraries\\vendor\\joomla\\database\\src\\Mysqli\\MysqliDriver.php on line 716, referer: http://localhost/joomla4dev/installation/index.php?view=setup&layout=default

```
From the log, the installation failed silently a long time ago and silently logged the error without informing in the browser.

The installer still running in the browser:

![screenshot-localhost-2018-03-08-17-29-30-809](https://user-images.githubusercontent.com/3242390/37156162-693e1fe0-22f6-11e8-9aaa-f4f29db19a20.png)


You might consider reopening this issue.

**Comment by zero-24** [comment]
Please increase that value for the max execution time. On your local device this can happen because our device is doing more than just running xampp and joomla (the same reason it happen on 3.x)

If you want you can also try 4.0 on a live server there this should not happen ;)

**Comment by emuthomi** [comment]
Are you aware that most shared hosting environments where Joomla is likely to be installed by everyday users limit the max_execution_time to 30 seconds? 

An example is http://support.hostgator.com/articles/php-settings-that-cannot-be-changed
and https://support.ehost.com/articles/cpanel/php-settings-that-cannot-be-changed

**Comment by mbabker** [comment]
30 second max execution time should be more than enough for any single task in the install app and the fact something is pushing it past that limitation means there's a problem somewhere.  This isn't a "fix your server config" issue.

**Comment by zero-24** [comment]
> Are you aware that most shared hosting environments where Joomla is likely to be installed by everyday users limit the max_execution_time to 30 seconds?

> 30 second max execution time should be more than enough for any single task in the install app and the fact something is pushing it past that limitation means there's a problem somewhere. This isn't a "fix your server config" issue.

Sure but server != local install. This is the reason i suggested to test on a live server to be sure.

**Comment by mbabker** [comment]
That does nothing but give fuel to the "fix your server config" argument.  I can run the installer just fine on my local AMP stack.  So it's not a matter of "it works on this host but not WAMP".

The posted data from the PHP configuration is more than ample to run Joomla regardless of where the physical system running the PHP binary is hosted.

**Comment by dgrammatiko** [comment]
> This screen will show things like the minimum php requirements and if every requirement is there before installation.

There is such a page, but appears only when something doesn't meet the prerequisites. I guess we could expand this with the proposed configuration min values (?)

**Comment by zero-24** [comment]
Feel free to reopen if you find a way to fix this issue. I would bet that this depends on local behaviors like what is also running on this PC in parallel (maybe something like PS), what IO / CPU / SSD or HHD is build in etc.

> I can run the installer just fine on my local AMP stack. So it's not a matter of "it works on this host but not WAMP".

Sure you have decent hardware and I guess your CPU / IO is not at its limits with your currently running software ;)

**Comment by mbabker** [comment]
> Sure you have decent hardware and I guess your CPU / IO is not at its limits with your currently running software ;)

Nope.  Two separate devices which are 4+ year old Apple hardware models and according to my RAM monitoring software typically over 60% of those resources are in use.  Although if I wanted to poison my gaming rig with a development environment that would probably work out better as far as the hardware platform goes, but I refuse to do any development work (paid or volunteer) on a Windows platform because I see the struggles my co-workers go through to use the required tooling on their Windows systems and how often the end result is "I can't get this to work, you do this since your stuff just works" 🤣

**Comment by ot2sen** [comment]
No problem in making latest nightly build install. Also worked fine two days ago. (have default 30 limit)
Just read this thread, went to developer.joomla.org and downloaded zip. All in 3 mins or so, works :)
![08-03-2018_4dev0803](https://user-images.githubusercontent.com/790445/37179030-9d459eac-2324-11e8-9628-de8c2b0474d6.png)

**Comment by zero-24** [comment]
Yes just as I said it depends on your machine ;) Just to be sure @ot2sen are you on windows or mac OS? So we can confirm a successful install on both :)

**Comment by ot2sen** [comment]
Win10 using Ampps :)

**Comment by brianteeman** [comment]
No problem for me


Setting | Value
-- | --
 
PHP Built On | Windows NT DESKTOP-VPRGVNL 10.0 build 16299 (Windows 10) AMD64
Database Version | 5.7.21-log
Database Collation | utf8mb4_unicode_ci
Database Connection Collation | utf8_general_ci
PHP Version | 7.2.2
Web Server | Apache/2.4.29 (Win64) mod_fcgid/2.3.9
WebServer to PHP Interface | cgi-fcgi
Joomla! Version | Joomla! 4.0.0-dev Development [ Amani ] 19-November-2017 16:34 GMT
User Agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.186 Safari/537.36

**Comment by emuthomi** [comment]
I tried today with Joomla 3.8.5 and the installation went on well, but not smoothly. It reached a point the installer somehow froze but after refreshing the page the installation completed in a second or two and I could enjoy my new J3.8 installation.

(In J4.0, refreshing the installer triggers errors showing missing tables in the database)

I decided to give J4.0 a try again. This time round I chose English US language and MySQL PDO as my db driver. 

First issue was that it seems with MySQL PDO selected I needed to create my db beforehand,

![screenshot-localhost-2018-03-09-09-23-38-096](https://user-images.githubusercontent.com/3242390/37194027-5a0983c0-237d-11e8-97b2-6d54cfbd923a.png)

I went and created my database. Tried to run the installer again, but it was complaining of some missing tables. So I wiped off the contents of the directory and got fresh files. 

15 minutes later:

```
[Fri Mar 09 09:28:51.256782 2018] [php7:error] [pid 3036:tid 1720] [client ::1:52400] PHP Fatal error:  Maximum execution time of 60 seconds exceeded in C:\\xampp71\\htdocs\\joomla40dev\\libraries\\vendor\\joomla\\database\\src\\Pdo\\PdoDriver.php on line 416, referer: http://localhost/joomla40dev/installation/index.php
```

I would have tried install J4 in my older Xampp which runs PHP 5.6 but J4 requires PHP 7.

**Comment by infograf768** [comment]
mysql (PDO) is broken atm in 4.0. Please use mysqli.

**Comment by mbabker** [comment]
> mysql (PDO) is broken atm in 4.0. Please use mysqli.

No, it's not.  The automatic create database logic is broken, not the driver in full.

**Comment by infograf768** [comment]
that is what i meant

**Comment by emuthomi** [comment]
Today the installation went on well and completed successfully. The new admin interface is a shock! Let me familiarize myself with it.  Is there a way to move the side nav to the top?

Thank you all for your help.

**Comment by emuthomi** [comment]
Also sample data didn't install. I saw an issue reporting the same.

**Comment by dgrammatiko** [comment]
>  Is there a way to move the side nav to the top?

NO!
